packages/ui/src/components/Button/index.tsx

```diff
@@ -315,48 +304,43 @@ const StyledButton = styled('button', {
   &:hover,
   &:focus {
     text-decoration: none;
-    ${({ target, theme }) =>
-      target === '_blank' ? `gap: ${theme.space['1']};` : ''}
   }
 
   ${sizeStyles}
   ${variantStyles}
   ${({ disabled, theme }) =>
     disabled &&
     `
-    cursor: default;
-    pointer-events: none;
-    color: ${theme.colors.neutral.textDisabled};
-    svg {
-      fill: ${theme.colors.neutral.textDisabled};
-    }
+      cursor: default;
+      pointer-events: none;
+      color: ${theme.colors.neutral.textDisabled};
+      svg {
+        fill: ${theme.colors.neutral.textDisabled};
+      }
     `}
 
   ${({ variant, disabled, theme }) =>
     variant !== 'link' &&
     disabled &&
     `
-    background-color: ${theme.colors.neutral.backgroundDisabled};
-    border-color: ${theme.colors.neutral.borderDisabled};
-    box-shadow: none;
+      background-color: ${theme.colors.neutral.backgroundDisabled};
+      border-color: ${theme.colors.neutral.borderDisabled};
+      box-shadow: none;
     `}
 
-  ${({ extend, icon }) =>
+  ${({ extend, theme }) =>
     extend &&
     css`
-      display: inline-flex;
-      & ${StyledContent} {
-        transition: max-width 450ms ease, padding 150ms ease, margin 150ms ease;
+      & ${InhibitedStack} {
+        transition: max-width 250ms ease, margin-left 250ms ease;
         max-width: 0;
-        margin-right: 0;
-        ${icon ? 'padding-right: 0;' : 'padding-left: 0;'};
+        margin-left: -${theme.space['1']}px; // To counteract parent gap
```

Comment:
Are these comments valid in emotion? Because a normal CSS comment should be:
```suggestion
        margin-left: -${theme.space['1']}px; /* To counteract parent gap */
```